app/javascript/packs/admin/convertUserIdsToUsernameInputs.js

```diff
@@ -1,108 +1,128 @@
 import { h, render } from 'preact';
-import { UsernameInput } from '../../shared/components/UsernameInput';
-import { UserStore } from '../../shared/components/UserStore';
+import { UsernameInput } from '@components/UsernameInput';
+import { UserStore } from '@components/UserStore';
 import '@utilities/document_ready';
 
-async function convertUserIdFieldToUsernameField(targetField, users, fetchUrl) {
+const CO_AUTHOR_FIELD_CLASS_NAME = '.js-coauthor_username_id_input';
+const USERNAME_FIELD_CLASS_NAME = '.js-username_id_input';
+
+function generateFetchUrl(ids) {
+  const searchParams = new URLSearchParams();
+  ids.forEach((id) => searchParams.append('ids[]', Number(id)));
+  return `/admin/member_manager/users.json?${searchParams}'`;
+}
+
+function extractUserIds(value) {
+  return value
+    .split(',')
+    .map((id) => id.replace(' ', ''))
+    .filter((id) => id !== '');
+}
+
+function getAllUserIdsFromTextFields(fieldClassNames) {
+  const uniqueIds = new Set();
+
+  fieldClassNames.forEach((className) => {
+    document.querySelectorAll(className).forEach((field) => {
+      extractUserIds(field.value).forEach((id) => uniqueIds.add(id));
+    });
+  });
+  return [...uniqueIds];
+}
+
+async function fetchUsers(ids) {
+  if (ids.length <= 0) {
+    return new UserStore();
+  }
+  return await UserStore.fetch(generateFetchUrl(ids));
+}
+
+const fetchSuggestions = async (term, searchOptions = {}) => {
+  const searchParams = new URLSearchParams();
+  searchParams.append('limit', 10);
+  searchParams.append('search', term);
+  const userStore = await UserStore.fetch(
+    `/admin/member_manager/users.json?${searchParams}`,
+  );
```

Comment:
Instead of fetching all users when the page is loaded, fetch a user list when an admin types some characters.